x-pack/plugins/discover_enhanced/public/actions/view_in_discover/explore_data_context_menu_action.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable max-classes-per-file */
+
+import { i18n } from '@kbn/i18n';
+import { Action } from '../../../../../../src/plugins/ui_actions/public';
+import { DiscoverStart } from '../../../../../../src/plugins/discover/public';
+import {
+  EmbeddableContext,
+  IEmbeddable,
+  ViewMode,
+} from '../../../../../../src/plugins/embeddable/public';
+import { StartServicesGetter } from '../../../../../../src/plugins/kibana_utils/public';
+import { CoreStart } from '../../../../../../src/core/public';
+import {
+  VisualizeEmbeddableContract,
+  VISUALIZE_EMBEDDABLE_TYPE,
+} from '../../../../../../src/plugins/visualizations/public';
+
+// TODO: Replace this logic with KibanaURL once it is available.
+// https://github.com/elastic/kibana/issues/64497
+class KibanaURL {
+  public readonly path: string;
+  public readonly appName: string;
+  public readonly appPath: string;
+
+  constructor(path: string) {
+    const match = path.match(/^.*\/app\/([^\/\/#]+)(.+)$/);
```

Comment:
This should also work (duplicate `\/`) in your character class
```suggestion
    const match = path.match(/^.*\/app\/([^\/#]+)(.+)$/);
```